run-tests.php

```diff
@@ -1958,22 +1958,22 @@ function run_test(string $php, $file, array $env): string
         $temp_dir = str_replace($temp_source, $temp_target, $temp_dir);
     }
 
-    $main_file_name = basename($file, 'phpt');
-
-    $diff_filename = $temp_dir . DIRECTORY_SEPARATOR . $main_file_name . 'diff';
-    $log_filename = $temp_dir . DIRECTORY_SEPARATOR . $main_file_name . 'log';
-    $exp_filename = $temp_dir . DIRECTORY_SEPARATOR . $main_file_name . 'exp';
-    $output_filename = $temp_dir . DIRECTORY_SEPARATOR . $main_file_name . 'out';
-    $memcheck_filename = $temp_dir . DIRECTORY_SEPARATOR . $main_file_name . 'mem';
-    $sh_filename = $temp_dir . DIRECTORY_SEPARATOR . $main_file_name . 'sh';
-    $temp_file = $temp_dir . DIRECTORY_SEPARATOR . $main_file_name . 'php';
-    $test_file = $test_dir . DIRECTORY_SEPARATOR . $main_file_name . 'php';
-    $temp_skipif = $temp_dir . DIRECTORY_SEPARATOR . $main_file_name . 'skip.php';
-    $test_skipif = $test_dir . DIRECTORY_SEPARATOR . $main_file_name . 'skip.php';
-    $temp_clean = $temp_dir . DIRECTORY_SEPARATOR . $main_file_name . 'clean.php';
-    $test_clean = $test_dir . DIRECTORY_SEPARATOR . $main_file_name . 'clean.php';
-    $preload_filename = $temp_dir . DIRECTORY_SEPARATOR . $main_file_name . 'preload.php';
-    $tmp_post = $temp_dir . DIRECTORY_SEPARATOR . $main_file_name . 'post';
+    $main_file_name = basename($file, '.phpt');
+
+    $diff_filename = $temp_dir . DIRECTORY_SEPARATOR . $main_file_name . '.diff';
+    $log_filename = $temp_dir . DIRECTORY_SEPARATOR . $main_file_name . '.log';
+    $exp_filename = $temp_dir . DIRECTORY_SEPARATOR . $main_file_name . '.exp';
+    $output_filename = $temp_dir . DIRECTORY_SEPARATOR . $main_file_name . '.out';
+    $memcheck_filename = $temp_dir . DIRECTORY_SEPARATOR . $main_file_name . '.mem';
+    $sh_filename = $temp_dir . DIRECTORY_SEPARATOR . $main_file_name . '.sh';
+    $temp_file = $temp_dir . DIRECTORY_SEPARATOR . $main_file_name . '.php';
+    $test_file = $test_dir . DIRECTORY_SEPARATOR . $main_file_name . '.php';
+    $temp_skipif = $temp_dir . DIRECTORY_SEPARATOR . $main_file_name . '.skip.php';
+    $test_skipif = $test_dir . DIRECTORY_SEPARATOR . $main_file_name . '.skip.php';
+    $temp_clean = $temp_dir . DIRECTORY_SEPARATOR . $main_file_name . '.clean.php';
+    $test_clean = $test_dir . DIRECTORY_SEPARATOR . $main_file_name . '.clean.php';
+    $preload_filename = $temp_dir . DIRECTORY_SEPARATOR . $main_file_name . '.preload.php';
+    $tmp_post = $temp_dir . DIRECTORY_SEPARATOR . $main_file_name . '.post';
```

Comment:
These changes look unrelated?

Reply:
They are related: to run a test from a test, I had to give it extension `.phpt.txt` to avoid them being picked up by run-tests directly. However, without these changes, filenames like `test.phpt.txtdiff` were generated.

Reply:
I see, that makes sense. Alternatively, it would be possible to create the `.phpt` file during test execution.

Reply:
Though in this particular case, I'm not sure the self-invoking test is even particularly useful: If the test weren't skipped, we'd be seeing a failure during a normal test run.

Reply:
Makes sense, I'll just nuke this test then:)